### ISSUE TITLE
Document outline can be incorrect when scopes have the same name

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#4388]
One repro is below
```

namespace ns4 {
	namespace ns4 {
		class ns3 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
		class ns32 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
	}
	namespace ns52 {
		class ns3 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
		class ns32 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
	}
}
namespace n45 {
	namespace ns5 {
		class ns3 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
		class ns32 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
	}
	namespace ns52 {
		class ns3 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
		class ns32 {
		public:
			int main()
			{
				return 0;
			}
			int func() { return 1; }
		};
	}
}
```

**Comment by sean-mcmanus** [comment]
This bug still repros. A simpler repro is 
```
namespace foo
{
	namespace foo {
		class foo {};
	}
}
```

**Comment by Colengms** [comment]
A fix for this is in 1.3.0.  Closing as a dupe of: https://github.com/microsoft/vscode-cpptools/issues/7024
